github/VisualStudio
# Adding Publish to Essentials extension

Add `Publish to GitHub` section to `Team Explorer - Synchronization` page:

![image](https://user-images.githubusercontent.com/11719160/60255862-6bbde900-98c8-11e9-990f-c4bf3c7143e6.png)

When `Publish to GitHub` button is clicked:

![image](https://user-images.githubusercontent.com/11719160/60255935-8db76b80-98c8-11e9-8a67-39bc6c4bb532.png)

When user opens the `Team Explorer - Synchronization` page, only `GitHub.VisualStudio.16` will be loaded:

![image](https://user-images.githubusercontent.com/11719160/60261022-cf4d1400-98d2-11e9-8b9a-4b26fb7fdc24.png)

### How to test

1. Ensure that only the GitHub Essentials extension is installed
2. Create a new project
3. `Add to Source Control > Git`
4. Click `Publish to GitHub` button
![image](https://user-images.githubusercontent.com/11719160/64859469-3badf880-d622-11e9-96bf-5d062285b457.png)
5. Choose a server (GitHub or GitHub Enterprise), owner (user or organization) and repository name
6. Check `Private` (if you don't want your test repository to be publicly visible)
7. Click `Publish` button
![image](https://user-images.githubusercontent.com/11719160/64859626-b119c900-d622-11e9-8e52-5906af12f8c0.png)
8. Expect `Repository created successfully` message
![image](https://user-images.githubusercontent.com/11719160/64859736-f938eb80-d622-11e9-9792-6241b4c28ea9.png)
9. Click on `Sync` button
10. Expect no publish sections to be visible
![image](https://user-images.githubusercontent.com/11719160/64859818-3604e280-d623-11e9-8cdc-e809a1866be4.png)
